From 9946158fac50d75737a8e26740cc38bcdb2fbf9f Mon Sep 17 00:00:00 2001
From: Dan Elphick <delphick@chromium.org>
Date: Thu, 22 Jul 2021 17:42:22 +0100
Subject: [PATCH] [Build] Reduce unnecessary includes of v8.h

Removes unnecessary includes of v8.h from
src/diagnostics/gdb-jit.h
src/diagnostics/system-jit-win.h
src/diagnostics/unwinder.h

by predeclaring types or using more appropriate headers.

Bug: v8:11879
Change-Id: I17f42acfef8e61133988453d67c3c0d473ff0337
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3045702
Auto-Submit: Dan Elphick <delphick@chromium.org>
Commit-Queue: Dan Elphick <delphick@chromium.org>
Reviewed-by: Ross McIlroy <rmcilroy@chromium.org>
Cr-Commit-Position: refs/heads/master@{#75870}
---
 src/diagnostics/arm/unwinder-arm.cc         | 3 +++
 src/diagnostics/arm64/unwinder-arm64.cc     | 2 ++
 src/diagnostics/gdb-jit.cc                  | 1 +
 src/diagnostics/gdb-jit.h                   | 5 +++--
 src/diagnostics/ia32/unwinder-ia32.cc       | 2 ++
 src/diagnostics/mips/unwinder-mips.cc       | 2 ++
 src/diagnostics/mips64/unwinder-mips64.cc   | 2 ++
 src/diagnostics/ppc/unwinder-ppc.cc         | 3 +++
 src/diagnostics/riscv64/unwinder-riscv64.cc | 2 ++
 src/diagnostics/s390/unwinder-s390.cc       | 3 +++
 src/diagnostics/system-jit-win.cc           | 1 +
 src/diagnostics/system-jit-win.h            | 5 +++--
 src/diagnostics/unwinder.cc                 | 2 ++
 src/diagnostics/unwinder.h                  | 5 ++---
 src/diagnostics/x64/unwinder-x64.cc         | 2 ++
 15 files changed, 33 insertions(+), 7 deletions(-)

diff --git a/src/diagnostics/arm/unwinder-arm.cc b/src/diagnostics/arm/unwinder-arm.cc
index 846bbfe6bc..e0e2f0e91f 100644
--- a/src/diagnostics/arm/unwinder-arm.cc
+++ b/src/diagnostics/arm/unwinder-arm.cc
@@ -2,7 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <memory>
+
 #include "include/v8-unwinder-state.h"
+#include "include/v8.h"
 #include "src/diagnostics/unwinder.h"
 #include "src/execution/frame-constants.h"
 
diff --git a/src/diagnostics/arm64/unwinder-arm64.cc b/src/diagnostics/arm64/unwinder-arm64.cc
index 5a92512a17..0314458005 100644
--- a/src/diagnostics/arm64/unwinder-arm64.cc
+++ b/src/diagnostics/arm64/unwinder-arm64.cc
@@ -6,6 +6,8 @@
 
 namespace v8 {
 
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 
diff --git a/src/diagnostics/gdb-jit.cc b/src/diagnostics/gdb-jit.cc
index 4125dfaa4c..53c29cfb24 100644
--- a/src/diagnostics/gdb-jit.cc
+++ b/src/diagnostics/gdb-jit.cc
@@ -8,6 +8,7 @@
 #include <memory>
 #include <vector>
 
+#include "include/v8.h"
 #include "src/api/api-inl.h"
 #include "src/base/bits.h"
 #include "src/base/hashmap.h"
diff --git a/src/diagnostics/gdb-jit.h b/src/diagnostics/gdb-jit.h
index e1bc852f0a..82f5ce892c 100644
--- a/src/diagnostics/gdb-jit.h
+++ b/src/diagnostics/gdb-jit.h
@@ -5,8 +5,6 @@
 #ifndef V8_DIAGNOSTICS_GDB_JIT_H_
 #define V8_DIAGNOSTICS_GDB_JIT_H_
 
-#include "include/v8.h"
-
 //
 // GDB has two ways of interacting with JIT code.  With the "JIT compilation
 // interface", V8 can tell GDB when it emits JIT code.  Unfortunately to do so,
@@ -25,6 +23,9 @@
 //
 
 namespace v8 {
+
+struct JitCodeEvent;
+
 namespace internal {
 namespace GDBJITInterface {
 #ifdef ENABLE_GDB_JIT_INTERFACE
diff --git a/src/diagnostics/ia32/unwinder-ia32.cc b/src/diagnostics/ia32/unwinder-ia32.cc
index 5a92512a17..0314458005 100644
--- a/src/diagnostics/ia32/unwinder-ia32.cc
+++ b/src/diagnostics/ia32/unwinder-ia32.cc
@@ -6,6 +6,8 @@
 
 namespace v8 {
 
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 
diff --git a/src/diagnostics/mips/unwinder-mips.cc b/src/diagnostics/mips/unwinder-mips.cc
index 5a92512a17..0314458005 100644
--- a/src/diagnostics/mips/unwinder-mips.cc
+++ b/src/diagnostics/mips/unwinder-mips.cc
@@ -6,6 +6,8 @@
 
 namespace v8 {
 
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 
diff --git a/src/diagnostics/mips64/unwinder-mips64.cc b/src/diagnostics/mips64/unwinder-mips64.cc
index 5a92512a17..0314458005 100644
--- a/src/diagnostics/mips64/unwinder-mips64.cc
+++ b/src/diagnostics/mips64/unwinder-mips64.cc
@@ -6,6 +6,8 @@
 
 namespace v8 {
 
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 
diff --git a/src/diagnostics/ppc/unwinder-ppc.cc b/src/diagnostics/ppc/unwinder-ppc.cc
index 43c6acb609..52c22221ff 100644
--- a/src/diagnostics/ppc/unwinder-ppc.cc
+++ b/src/diagnostics/ppc/unwinder-ppc.cc
@@ -3,6 +3,9 @@
 // found in the LICENSE file.
 #include "src/diagnostics/unwinder.h"
 namespace v8 {
+
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 }  // namespace v8
diff --git a/src/diagnostics/riscv64/unwinder-riscv64.cc b/src/diagnostics/riscv64/unwinder-riscv64.cc
index ccfb9268ea..84d2e41cfc 100644
--- a/src/diagnostics/riscv64/unwinder-riscv64.cc
+++ b/src/diagnostics/riscv64/unwinder-riscv64.cc
@@ -6,6 +6,8 @@
 
 namespace v8 {
 
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 
diff --git a/src/diagnostics/s390/unwinder-s390.cc b/src/diagnostics/s390/unwinder-s390.cc
index 43c6acb609..52c22221ff 100644
--- a/src/diagnostics/s390/unwinder-s390.cc
+++ b/src/diagnostics/s390/unwinder-s390.cc
@@ -3,6 +3,9 @@
 // found in the LICENSE file.
 #include "src/diagnostics/unwinder.h"
 namespace v8 {
+
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 }  // namespace v8
diff --git a/src/diagnostics/system-jit-win.cc b/src/diagnostics/system-jit-win.cc
index 120020597a..c77c223183 100644
--- a/src/diagnostics/system-jit-win.cc
+++ b/src/diagnostics/system-jit-win.cc
@@ -4,6 +4,7 @@
 
 #include "src/diagnostics/system-jit-win.h"
 
+#include "include/v8.h"
 #include "src/api/api-inl.h"
 #include "src/base/lazy-instance.h"
 #include "src/base/logging.h"
diff --git a/src/diagnostics/system-jit-win.h b/src/diagnostics/system-jit-win.h
index dffd34df6c..68410079bb 100644
--- a/src/diagnostics/system-jit-win.h
+++ b/src/diagnostics/system-jit-win.h
@@ -5,9 +5,10 @@
 #ifndef V8_DIAGNOSTICS_SYSTEM_JIT_WIN_H_
 #define V8_DIAGNOSTICS_SYSTEM_JIT_WIN_H_
 
-#include "include/v8.h"
-
 namespace v8 {
+
+struct JitCodeEvent;
+
 namespace internal {
 namespace ETWJITInterface {
 void Register();
diff --git a/src/diagnostics/unwinder.cc b/src/diagnostics/unwinder.cc
index 1dd122a118..68ff679595 100644
--- a/src/diagnostics/unwinder.cc
+++ b/src/diagnostics/unwinder.cc
@@ -6,6 +6,8 @@
 
 #include <algorithm>
 
+#include "include/v8.h"
+#include "src/execution/frame-constants.h"
 #include "src/execution/pointer-authentication.h"
 
 namespace v8 {
diff --git a/src/diagnostics/unwinder.h b/src/diagnostics/unwinder.h
index 4cad2897fd..893415a813 100644
--- a/src/diagnostics/unwinder.h
+++ b/src/diagnostics/unwinder.h
@@ -5,12 +5,11 @@
 #ifndef V8_DIAGNOSTICS_UNWINDER_H_
 #define V8_DIAGNOSTICS_UNWINDER_H_
 
-#include "include/v8.h"
-#include "src/common/globals.h"
+#include "include/v8-internal.h"
 
 namespace v8 {
 
-i::Address Load(i::Address address);
+internal::Address Load(internal::Address address);
 
 }  // namespace v8
 
diff --git a/src/diagnostics/x64/unwinder-x64.cc b/src/diagnostics/x64/unwinder-x64.cc
index 5a92512a17..0314458005 100644
--- a/src/diagnostics/x64/unwinder-x64.cc
+++ b/src/diagnostics/x64/unwinder-x64.cc
@@ -6,6 +6,8 @@
 
 namespace v8 {
 
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 
-- 
2.35.1

